docs/Promise.md

```diff
@@ -0,0 +1,96 @@
+# Promise
+
+An implementation of the [ES2015 Promise specification](http://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects).
+
+Promises are used to reduce complications of asynchronous callbacks.
+
+## Creating Promises
+
+You can create a promise using the constructor:
+
+```typescript
+const p = new Promise((resolve, reject) => {
+  // some long running task
+  resolve();
+});
+```
+
+or by using static convenience methods:
+
+```typescript
+const resolvePromise = Promise.resolve();
+const rejectedPromise = Promise.reject();
+```
+
+Values can be wrapped in a resolved promise to facilitate things
+like simple caching:
```

Comment:
Don't like "things like..." perhaps to facilitate simple caching and other <insert word that describes the right pattern>